### ISSUE TITLE
Enum Action Parameters Incorrectly Converted to Integer Value

### ISSUE THREAD

**Issue by AlexaHLE** [object/Simple.OData.Client#210]
When an action requires a parameter that is an enum type, it appears that the parameter is incorrectly converted to its int value instead of its string name, and then the operation fails before the request is made because the parameter is not the type specified in the metadata.

I have created a sample WebApi OData Service  with the following metadata
```
<?xml version="1.0" encoding="utf-8"?>
<edmx:Edmx Version="4.0" xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx">
    <edmx:DataServices>
        <Schema Namespace="Models" xmlns="http://docs.oasis-open.org/odata/ns/edm">
            <EntityType Name="Entity">
                <Key>
                    <PropertyRef Name="Id" />
        
                </Key>
                <Property Name="Id" Type="Edm.Int32" Nullable="false" />
                <Property Name="Name" Type="Edm.String" />
                <Property Name="Rank" Type="Models.Rank" Nullable="false" />
      
            </EntityType>
            <ComplexType Name="EnumClass">
                <Property Name="Rank" Type="Models.Rank" Nullable="false" />
                <Property Name="Name" Type="Edm.String" />
      
            </ComplexType>
            <EnumType Name="Rank">
                <Member Name="First" Value="0" />
                <Member Name="Second" Value="1" />
                <Member Name="Third" Value="2" />
      
            </EnumType>
    
        </Schema>
        <Schema Namespace="Test" xmlns="http://docs.oasis-open.org/odata/ns/edm">
            <Action Name="MakeFromClass" IsBound="true">
                <Parameter Name="bindingParameter" Type="Collection(Models.Entity)" />
                <Parameter Name="args" Type="Models.EnumClass" />
                <ReturnType Type="Models.Entity" />
      
            </Action>
            <Action Name="MakeFromParam" IsBound="true">
                <Parameter Name="bindingParameter" Type="Collection(Models.Entity)" />
                <Parameter Name="Name" Type="Edm.String" Unicode="false" />
                <Parameter Name="Rank" Type="Models.Rank" Nullable="false" />
                <ReturnType Type="Models.Entity" />
      
            </Action>
            <EntityContainer Name="Container">
                <EntitySet Name="Entities" EntityType="Models.Entity" />
      
            </EntityContainer>
    
        </Schema>
  
    </edmx:DataServices>

</edmx:Edmx>
```

I have verfied in Postman that the actions work correctly with the following requests
```
POST http://localhost/ODataEnum/odata/Entities/Test.MakeFromParam
{
    "Name" : "Entity Name",
    "Rank" :"Second"
}
```
and
``` 
POST http://localhost/ODataEnum/odata/Entities/Test.MakeFromClass
{
    "args" : {
        "Name" : "Entity Name",
        "Rank" : "Third"
    }
}
```

However, Using ODataClient yields the following errors:
```
Entity result = await this.client.For<Entity>()
                .Action("MakeFromParam")
                .Set(new
                {
                    Name = "Entity Name",
                    Rank = Rank.Second
                })
                .ExecuteAsSingleAsync();
```

> Microsoft.OData.Core.ODataException: A primitive value was specified; however, a value of the non-primitive type 'Models.Rank' was expected.

This occurs after the client requests the metadata, but before the action request is made.

and
```
numClass args = new EnumClass
            {
                Name = "Entity Name",
                Rank = Rank.Third
            };

            Entity result = await this.client.For<Entity>()
                .Action("MakeFromClass")
                .Set(new
                {
                    args = args
                })
                .ExecuteAsSingleAsync();
```
> Microsoft.OData.Core.ODataException: The value for parameter 'args' is of type 'Models.EnumClass'. WriteValue can only write null, ODataComplexValue, ODataEnumValue and primitive types that are not Stream type.

I'm not sure this error is related to the enum parameter, it was just an attempt to force correct type. However, from my understanding it should work since EnumClass is a Complex type on my service.

by Altering the service configuration to expect an integer Rank parameter, MakeFromParam is successful. I have also tried configuring it to expect a string Rank, which causes a similar error
> Microsoft.OData.Core.ODataException: An incompatible primitive type 'Edm.Int32[Nullable=False]' was found for an item that was expected to be of type 'Edm.String[Nullable=True]

Is there a way to configure the client to convert enum types to their names instead of values for Actions?

**Comment by object** [comment]
Thank you for so well described issue report. I will investigate this and come back to you.

**Comment by object** [comment]
Fixed in version 4.17.2.
